sqlglot/tokens.py

```diff
@@ -777,6 +779,27 @@ def _scan_number(self):
             else:
                 return self._add(TokenType.NUMBER)
 
+    def _scan_hex_literal(self):
+        prefix = "0x"
+        text = prefix
+        self._advance()
+
+        while True:
+            if not self._peek:
+                if text == prefix:
+                    return self._add(TokenType.IDENTIFIER, text)
```

Comment:
i don't think this is correct,

if you de select 0x, it should actually be select 0 AS x

Reply:
Based on:
```
mysql> select @@version;
+-----------+
| @@version |
|-----------|
| 8.0.22-13 |
+-----------+
mysql> select 0x;
(1054, "Unknown column '0x' in 'field list'")
mysql> select `0x`;
(1054, "Unknown column '0x' in 'field list'")
mysql> select 0x from table_that_exists;
(1054, "Unknown column '0x' in 'field list'")
mysql> select `0x` from table_that_exists;
(1054, "Unknown column '0x' in 'field list'")
```

Reply:
But if another database tokenizes it differently … that's tricky.

Reply:
postgres: 0 as x
duckdb: 0 as x
sqlite3: Parse error: unrecognized token: "0x"
bigquery: Request contains an invalid argument

Reply:
postgres handles select 0x as select 0 as x which is more correct to me as you can't have ro shouldn't have identifiers prefixed with a 0. so i think it's ok to stick with postgres's interpretation on this one

Reply:
in general 0x should probably be considered user error, but i think i'd prefer to do 0 as x

Reply:
Right.  It's such a trivial edge case, and always going to be a mistake.  Although — the Postgres interpretation is influenced by the context that Postgres does not support this form of literal.